src/sequences/SequenceRandom.ts

```diff
@@ -0,0 +1,77 @@
+import { ValidationStatus } from '@/shared/ValidationStatus';
+import { SequenceParamsSchema, SequenceExportModule,
+         SequenceExportKind } from './SequenceInterface';
+import { SequenceCached } from './SequenceCached';
+
+/**
+ *
+ * @class SequenceClassRandom
+ * Creates a sequence of random integers in a specified range.
+ *
+ */
+class SequenceRandom extends SequenceCached {
+	name = "Random Integers in Range";
+	description = "A sequence of integers chosen indepenently uniformly from n to m inclusive.";
+	params: SequenceParamsSchema[] = [new SequenceParamsSchema(
+		'min',
+		'number',
+		'Minimum value attainable',
+		false,
+		0 
+	),
+	new SequenceParamsSchema(
+		'max',
+		'number',
+		'Maximum value attainable',
+		false,
+		9
+	)];
+
+	private minimum = 0;
+	private maximum = 0;
+
+	/**
+     	*Creates an instance of SequenceRandom
+     	* @param {*} ID the ID of the sequence
+	*/
+	constructor (ID: number) {
+		super(ID);
+	}
+
+	validate(){
+		this.settings['name'] = 'Constant';
+		const superStatus = super.validate();
+		if (!superStatus.isValid) {
+			return superStatus;
+		}
+
+		if (this.settings['min'] == undefined || this.settings['max'] == undefined ) {
+			this.isValid = false;
+			return new ValidationStatus(false, ["The min or max parameter is missing."]);
+		}
+
+		if ( !Number.isInteger( Number(this.settings.min) ) ) return new ValidationStatus(false, ["The min value must be an integer."]);
+		if ( !Number.isInteger( Number(this.settings.max) ) ) return new ValidationStatus(false, ["The max value must be an integer."]);
+
+		this.isValid = true;
+		this.name = "Random integers " + String(Number(this.settings.min)) + " to " + String(Number(this.settings.max));
+
+		return new ValidationStatus(true);
+
+	}
+
+	calculate(n: number) {
+		this.minimum = Number(this.settings.min);
+		this.maximum = Number(this.settings.max);
+
```

Comment:
You should set `this.minimum` and `this.maximum` once in validate(), where in fact you are already doing the conversion to Number anyway, rather than re-doing the conversion on every call to calculate().

Reply:
Makes sense. General question:  where does one do general setup for a more general sequence?  validate() seems too special-purpose, at least based on its name, but sequences don't have a general setup() function that would only get called once.  Should that be an issue to add?

Reply:
There is already an initialize() method in SequenceInterface that is guaranteed to be called once per sequence. But it is called _after_ validate(), and by default checks that validation succeeded. But in the case of your two parameters, they already have to be parsed in validate(), so there is no reason not to set the object properties right in validate() as you have done. On the other hand, I agree, if you had to say pre-compute some big matrix or something like that one time before calculate() (or getElement(), in case of an uncached Sequence) was called, you would do that in initialize() rather than validate().